src/modules/core/components/IndexModal/IndexModalItem.tsx

```diff
@@ -1,31 +1,44 @@
 import React from 'react';
 import { FormattedMessage, defineMessages } from 'react-intl';
+
 import Icon from '~core/Icon';
 import Paragraph from '~core/Paragraph';
-import styles from './IndexModalItem.css';
+import { Tooltip } from '~core/Popover';
+
 import { getMainClasses } from '~utils/css';
+import { ItemShape } from './IndexModal';
+
+import styles from './IndexModalItem.css';
 
 interface Props {
-  title: object;
-  description: object;
-  comingSoon?: boolean;
-  icon: string;
-  onClick?: () => void;
+  item: ItemShape;
 }
 
 const MSG = defineMessages({
   coming: {
-    id: 'core.IndexModal.IndexModalItem.coming',
+    id: 'IndexModal.IndexModalItem.coming',
     defaultMessage: 'Coming soon',
   },
+  permissionsMessageFallback: {
+    id: 'IndexModal.IndexModalItem.permissionsMessageFallback',
+    defaultMessage: `You must have the required permissions in the
+      relevant domains, in order to take this action`,
+  },
 });
 
+const displayName = 'IndexModal.IndexModalItem';
+
 const IndexModalItem = ({
-  title,
-  description,
-  icon,
-  onClick,
-  comingSoon,
+  item: {
+    title,
+    description,
+    icon,
+    onClick,
+    comingSoon = false,
+    permissionRequired = false,
```

Comment:
not sure if those are needed, if something wasn't passed, it will be undefined so !comingSoon will give false either way

Reply:
Sure, you're right. 

I just have this habit of initializing all boolean values manually :)